### PR TITLE
Use a thread safe class for holding header values

### PR DESCRIPTION
**Pull request by jonmv**
@bjorncs

### CODE REVIEW

**Review by bjorncs — APPROVED** [review]

File: jdisc_core/src/main/java/com/yahoo/jdisc/HeaderFields.java [hunk 1]
```diff
@@ -14,13 +19,7 @@
...
 public class HeaderFields implements Map<String, List<String>> {
 
-    private final TreeMap<String, List<String>> content = new TreeMap<>(new Comparator<String>() {
-
-        @Override
-        public int compare(String lhs, String rhs) {
-            return lhs.compareToIgnoreCase(rhs);
-        }
-    });
+    private final ConcurrentSkipListMap<String, List<String>> content = new ConcurrentSkipListMap<>(String::compareToIgnoreCase);
```
Comment (baldersheim) [inline jdisc_core/src/main/java/com/yahoo/jdisc/HeaderFields.java:22]: Won't this kill performance for high qps applications ?

Comment (jonmv) [reply jdisc_core/src/main/java/com/yahoo/jdisc/HeaderFields.java:22]: How many headers are there?

Comment (jonmv) [reply jdisc_core/src/main/java/com/yahoo/jdisc/HeaderFields.java:22]: We could synchronize the methods instead, I guess.

Comment (jonmv) [reply jdisc_core/src/main/java/com/yahoo/jdisc/HeaderFields.java:22]: Can't use a `syncrhonizedMap`, as this class implements `Map` itself, and exposes all views of the underlying map.

Comment (jonmv) [reply jdisc_core/src/main/java/com/yahoo/jdisc/HeaderFields.java:22]: Ah, there are unsynchronized `ArrayList` objects in the map, so need to do something about them as well.

**Pull request merged**